classes/external/get_data_merge_requests.php

```diff
@@ -0,0 +1,198 @@
+                                            'removeuservalue' => $removeuservalue,
+                                            'removeuserid' => $removeuserid,
+                                            'keepuserfield' => $keepuserfield,
+                                            'keepuservalue' => $keepuservalue,
+                                            'keepuserid' => $keepuserid,
+                                            'id' => $id,
+                                            'status' => $status]);
+        $sql = "SELECT
+                    id, removeuserfield, removeuservalue, removeuserid,
+                    keepuserfield, keepuservalue, keepuserid,
+                    timeadded, timemodified, status, retries, log
+                FROM
+                   {" .merge_request::TABLE_MERGE_REQUEST. "}
+                ";
+        $paramsquery = array();
+        if (isset($removeuserfield) && !empty($removeuserfield)) {	
+			[$insql, $params] = $DB->get_in_or_equal($removeuserfield);
+			$whereclauses[] = 'removeuserfield '.$insql; 
+            array_push($paramsquery, $inparams);
+        }
+        if (isset($removeuservalue) && !empty($removeuservalue)) {
+            [$insql, $params] = $DB->get_in_or_equal($removeuservalue);
+			$whereclauses[] = 'removeuservalue '.$insql; 
+            array_push($paramsquery, $inparams);
+        }
+        if (isset($removeuserid) && !empty($removeuserid)) {
+			[$insql, $params] = $DB->get_in_or_equal($removeuserid);
+			$whereclauses[] = 'removeuserid '.$insql; 
+            array_push($paramsquery, $inparams);
+        }
+        if (isset($keepuserfield) && !empty($keepuserfield)) {
+			array_push($paramsquery, $keepuserfield);            
+			[$insql, $params] = $DB->get_in_or_equal($keepuserfield);
+			$whereclauses[] = 'keepuserfield '.$insql;
+            array_push($paramsquery, $inparams); 
+        }
+        if (isset($keepuservalue) && !empty($keepuservalue)) {
+			[$insql, $params] = $DB->get_in_or_equal($keepuservalue);
+			$whereclauses[] = 'keepuservalue '.$insql; 
+            array_push($paramsquery, $inparams);
+        } 
+        if (isset($keepuserid) && !empty($keepuserid)) {
+            [$insql, $inparams] = $DB->get_in_or_equal($keepuserid);
+			$whereclauses[] = 'keepuserid '.$insql; 
+            array_push($paramsquery, $inparams);
+        }
+        if (isset($id) && !empty($id)) {
+         	[$insql, $inparams] = $DB->get_in_or_equal($id);
+			$whereclauses[] = 'id '.$insql; 
+            array_push($paramsquery, $inparams);
+        }
+        if (isset($status) && !empty($status)) {
+         	[$insql, $inparams] = $DB->get_in_or_equal($status);
+			$whereclauses[] = 'status ' .$insql; 
+            array_push($paramsquery, $inparams);
+        }
+        if (count($whereclauses) > 0) {
+            $sql .= ' WHERE ' . implode(' AND ', $whereclauses);
+        }
+        return $DB->get_records_sql($sql, $paramsquery);
```

Comment:
when I run this task I received an error 
`    "debuginfo": "ERROR:  invalid input syntax for type bigint: \"Array\"\nSELECT\n                    id, removeuserfield, removeuservalue, removeuserid,\n                    keepuserfield, keepuservalue, keepuserid,\n                    timeadded, timemodified, status, retries, log\n                FROM\n                   mdl_tool_mergeusers_queue\n                 WHERE keepuserid = $1 AND id = $2\n-- line 140 of /admin/tool/mergeusers/classes/external/get_data_merge_requests.php: call to pgsql_native_moodle_database->get_records_sql()\n[array (\n  0 => 1,\n  1 => \n  array (\n    0 => 16,\n  ),\n)]"`

Reply:
in postman I passed the following parameters
keepuserid -> 3
and 
id -> 16

Reply:
I solved it using pushing the value and not an array in the variable $paramsquery used in sql (return $DB->get_records_sql($sql, $paramsquery);)

 array_push($paramsquery, $id);